tests/src/JIT/Performance/CodeQuality/Span/SpanBench.cs

```diff
@@ -364,12 +445,7 @@ public static void BubbleSortArray()
         [InlineData(1000)]
         public static void TestSpanConstructorByte(int length)
         {
-            var array = new byte[length];
-            Span<byte> span;
-            foreach (var iteration in Benchmark.Iterations)
-                using (iteration.StartMeasurement())
-                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
-                        span = new Span<byte>(array);
+            InvokeTestSpanConstructor<byte>(length);
```

Comment:
I just learned (from @KrzysztofCwalina) how you could  run benchmark tests with Type passed in as InlineData and avoid having multiple type specific definitions (like TestSpanConstructorByte and TestSpanConstructorString). In essence, have generic benchmark methods.

So, instead of having:
```C#
...
public static void TestSpanConstructorByte(int length)
...
public static void TestSpanConstructorString(int length)
```
You could do this:
```C#
[Benchmark(InnerIterationCount = 100000)]
[InlineData(1, typeof(byte))]
[InlineData(10, typeof(byte))]
[InlineData(100, typeof(byte))]
[InlineData(1000, typeof(byte))]
[InlineData(1, typeof(string))]
[InlineData(10, typeof(string))]
[InlineData(100, typeof(string))]
[InlineData(1000, typeof(string))]
public static void TestSpanConstructor(int length, Type type)
{
    MethodInfo method = typeof(MemoryTests).GetMethod("InvokeTestSpanConstructor").MakeGenericMethod(type);
    object[] parameters = { length };
    method.Invoke(null, parameters);
    /*foreach (var iteration in Benchmark.Iterations)
    {
        using (iteration.StartMeasurement())
        {
            for (int i = 0; i < Benchmark.InnerIterationCount; i++)
                method.Invoke(null, parameters);
        }
    }*/
}
public static void InvokeTestSpanConstructor<T>(int length)
{
    var array = new T[length];
    Invoke((int innerIterationCount) => TestSpanConstructor<T>(array, innerIterationCount, false),
"TestSpanConstructor<{0}>({1})", typeof(T).Name, length);
}
```

@JosephTremoulet, do you refactoring the tests this way is a good idea?